src/ReplicatedStorage/Client/InstanceComponents/Misc/FallDamage.lua

```diff
@@ -0,0 +1,46 @@
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local FallDamageConfig = require(ReplicatedStorage.Common.Config.FallDamage)
+local Trove = require(ReplicatedStorage.Packages.Trove)
+
+local FallDamage = {}
+FallDamage.__index = FallDamage
+
+function FallDamage:_trackCharacterFall(character: Model, humanoid: Humanoid)
+    local initialHeight = nil
+    local fallHeight = nil
+
+    self._trove:Connect(humanoid.FreeFalling, function(active)
+        if active then
+            initialHeight = character.HumanoidRootPart.Position.Y
+        else
+			fallHeight = initialHeight - character.HumanoidRootPart.Position.Y
```

Comment:
Something wacky is happening here with whitespace. Try converting spaces to tabs first.